src/engine/SpatialJoin.cpp

```diff
@@ -137,6 +107,30 @@ bool SpatialJoin::isConstructed() const {
   return false;
 }
 
+// ____________________________________________________________________________
+std::optional<long long> SpatialJoin::getMaxDist() const {
+  auto visitor = []<typename T>(const T& config) -> std::optional<long long> {
+    if constexpr (std::is_same_v<T, MaxDistanceConfig>) {
+      return config.maxDist_;
+    } else if constexpr (std::is_same_v<T, NearestNeighborsConfig>) {
+      return config.maxDist_;
+    }
+  };
+  return std::visit(visitor, config_);
+}
+
+// ____________________________________________________________________________
+std::optional<long long> SpatialJoin::getMaxResults() const {
+  auto visitor = []<typename T>(const T& config) -> std::optional<long long> {
+    if constexpr (std::is_same_v<T, MaxDistanceConfig>) {
+      return std::nullopt;
+    } else if constexpr (std::is_same_v<T, NearestNeighborsConfig>) {
+      return config.maxResults_;
```

Comment:
In the else case, use
` } else { static_assert(is_same_v<.....>);` 

To not have undefined behavior
if you ever change the types contained in the variant (e.g. add one).